app/models/user.rb

```diff
@@ -26,31 +31,35 @@ def filter(query)
       q = "%#{query}%"
       conditions = t[:title].matches(q)
       columns = %i(
-        grant_number identifier description principal_investigator data_contact 
+        grant_number identifier description principal_investigator data_contact
       )
       columns = ['grant_number', 'identifier', 'description', 'principal_investigator', 'data_contact']
       columns.each {|col| conditions = conditions.or(t[col].matches(q)) }
       self.where(conditions)
     end
   end
-  
+
   has_many :user_identifiers
   has_many :identifier_schemes, through: :user_identifiers
 
   ##
   # Possibly needed for active_admin
   #   -relies on protected_attributes gem as syntax depricated in rails 4.2
   #accepts_nested_attributes_for :roles
-  #attr_accessible :password_confirmation, :encrypted_password, :remember_me, 
-  #                :id, :email, :firstname, :last_login,:login_count, :orcid_id, 
-  #                :password, :shibboleth_id, :user_status_id, :surname, 
-  #                :user_type_id, :org_id, :skip_invitation, :other_organisation, 
+  #attr_accessible :password_confirmation, :encrypted_password, :remember_me,
+  #                :id, :email, :firstname, :last_login,:login_count, :orcid_id,
+  #                :password, :shibboleth_id, :user_status_id, :surname,
+  #                :user_type_id, :org_id, :skip_invitation, :other_organisation,
   #                :accept_terms, :role_ids, :dmponline3, :api_token,
-  #                :organisation, :language, :language_id, :org, :perms, 
+  #                :organisation, :language, :language_id, :org, :perms,
   #                :confirmed_at, :org_id
 
   validates :email, email: true, allow_nil: true, uniqueness: {message: _("must be unique")}
 
+
+  validates :prefs, presence: true
+  before_validation :create_default_preferences
```

Comment:
this can probably go away once we start saving only the deviations from default.
We've found that validate? always fires before save (which gets fired after create and update). Since we're validating the presence_of, before_create never fires since the validation step fails.
Going to leave for now. We'll clean up in the new ticket #481